test/operators_test.cpp

```diff
@@ -932,5 +932,31 @@ main()
 
     cout << "Performed tests on MyLongInt objects.\n";
 
+// Where C++11 constexpr is available, compile-time test some of the operators that are able to use it to propagate constexpr
+#ifndef BOOST_NO_CXX11_CONSTEXPR
+#if !defined(BOOST_MSVC) || (_MSC_VER >= 1922)
+    static_assert( ! static_cast<bool>( MyInt{ 1 } == MyInt{ 2 } ), "" );
+    static_assert(                      MyInt{ 1 } != MyInt{ 2 },   "" );
+    static_assert(                      MyInt{ 1 } <  MyInt{ 2 },   "" );
+    static_assert(                      MyInt{ 1 } <= MyInt{ 2 },   "" );
+    static_assert( ! static_cast<bool>( MyInt{ 1 } >  MyInt{ 2 } ), "" );
+    static_assert( ! static_cast<bool>( MyInt{ 1 } >= MyInt{ 2 } ), "" );
+
+    static_assert( ! static_cast<bool>( MyInt{ 2 } == MyInt{ 1 } ), "" );
+    static_assert(                      MyInt{ 2 } != MyInt{ 1 },   "" );
+    static_assert( ! static_cast<bool>( MyInt{ 2 } <  MyInt{ 1 } ), "" );
+    static_assert( ! static_cast<bool>( MyInt{ 2 } <= MyInt{ 1 } ), "" );
+    static_assert(                      MyInt{ 2 } >  MyInt{ 1 },   "" );
+    static_assert(                      MyInt{ 2 } >= MyInt{ 1 },   "" );
+
+    static_assert(                      MyInt{ 1 } == MyInt{ 1 },   "" );
+    static_assert( ! static_cast<bool>( MyInt{ 1 } != MyInt{ 1 } ), "" );
+    static_assert( ! static_cast<bool>( MyInt{ 1 } <  MyInt{ 1 } ), "" );
+    static_assert(                      MyInt{ 1 } <= MyInt{ 1 },   "" );
+    static_assert( ! static_cast<bool>( MyInt{ 1 } >  MyInt{ 1 } ), "" );
+    static_assert(                      MyInt{ 1 } >= MyInt{ 1 },   "" );
+#endif
+#endif
+
```

Comment:
Ideally these should go into a separate test source file (e.g. `operators_constexpr_test.cpp`) so that any defects in implementations that do not support constexpr (or if Boost.Config gets C++11 constexpr detection wrong on certain implementations) that the failures don't fail the current `operators_test.cpp` run.